src/scripts/create_fullstack_ner.py

```diff
@@ -0,0 +1,180 @@
+# /// script
+# requires-python = ">=3.10,<4.0"
+# dependencies = [
+#     "datasets==3.5.0",
+#     "huggingface-hub==0.24.0",
+#     "pandas==2.2.0",
+#     "requests==2.32.3",
+# ]
+# ///
+
+"""Create and upload the FullStack-LV-mini NER dataset from CoNLL-U format."""
+
+import glob
+import os
+from collections import defaultdict
+from typing import Dict, List, Union
+
+import pandas as pd
+from datasets import Dataset, DatasetDict, Split
+from huggingface_hub import HfApi
+from requests import HTTPError
+
+
+def parse_conllu_data(raw_data: str) -> List[Dict[str, Union[List[str], str]]]:
+    """Parse CoNLL-U format data and return a list of sentence records."""
+    records = []
+    lines = raw_data.strip().split("\n")
+
+    # Initialize data dictionary for current sentence
+    data_dict: Dict[str, List[str]] = defaultdict(list)  # type: ignore[assignment]
+
+    for line in lines:
+        line = line.strip()
+
+        # Skip comments (lines starting with #)
+        if line.startswith("#"):
+            continue
+
+        # Empty line indicates end of sentence
+        elif line == "":
+            if len(data_dict["tokens"]) > 0:
+                # Create record with tokens, labels, and text
+                record = {
+                    "tokens": data_dict["tokens"],
+                    "labels": data_dict["labels"],
+                    "text": " ".join(data_dict["tokens"]),
+                }
+                records.append(record)
+            # Reset for next sentence
+            data_dict = defaultdict(list)
+
+        # Parse token line
+        else:
+            columns = line.split("\t")
+            if len(columns) >= 7:  # Ensure we have enough columns
+                token_id = columns[0]
+                token = columns[1]
+                # Use column 6 for NER tags (0-indexed, so columns[6] is column 7)
+                ner_tag = columns[6] if len(columns) > 6 else "O"
+
+                # Skip multi-word tokens (those with "-" in ID)
+                if "-" not in token_id and "." not in token_id:
+                    data_dict["tokens"].append(token)
+                    data_dict["labels"].append(ner_tag)
+
+    # Handle last sentence if data doesn't end with empty line
+    if len(data_dict["tokens"]) > 0:
+        record = {
+            "tokens": data_dict["tokens"],
+            "labels": data_dict["labels"],
+            "text": " ".join(data_dict["tokens"]),
+        }
+        records.append(record)
+
+    return records  # type: ignore[return-value]
+
+
+def load_fullstack_data() -> List[Dict[str, Union[List[str], str]]]:
+    """Load and parse all FullStack NER data from the local directory."""
+    # Path to the local data directory
+    data_dir = "../FullStack/NamedEntities/data"
```

Comment:
This seems very specific. Why not just clone the repo here in the code if it doesn't exist?